src/v3/src/components/PasswordRequirements/PasswordRequirements.tsx

```diff
@@ -39,22 +37,23 @@ const PasswordRequirements: UISchemaElementComponent<{
 }> = ({ uischema }) => {
   const { data } = useWidgetContext();
   const {
-    fieldKey,
     header,
     id,
     requirements,
     settings,
     userInfo,
     validationDelayMs,
   } = uischema.options as PasswordRequirementsElement['options'];
-  const password = get(data, fieldKey);
+  const password = 'credentials.newPassword' in data
```

Comment:
why depend on data bag in transformer layer, those preset data logics have been removed. 

also seems like we have testing gap for this scenario

Reply:
> why depend on data bag in transformer layer, those preset data logics have been removed.
do you mean in places where we are setting a default value, like identifier? 

> also seems like we have testing gap for this scenario
I will update the password expired test to capture the requirement validation hook, the reason I didnt do it initially was because I wasnt sure if we were going to remove that valdiation from the app, I know we talked about it during bug bash review but wanted to wait until they decided, but I will add it and we can just remove the test if that happens.

Reply:
> do you mean in places where we are setting a default value, like identifier?

yep, those logic have been removed. and seems like you are doing refactor to remove fieldKey in schema, right? then maybe add a todo comment, and revisit once decision is made.

Reply:
Ah, makes sense, I will do that and link the bug ticket. 

How are we supposed to set default data from the transformer? There are a few scenarios where we need to do that, in this case, for setting the idenfier into the input field when it is available in the cookie?

Reply:
that should not be changed, still touch formbag.data directly. but other transformation logic should depend on transaction object instead of the data bag

Reply:
Ahh, ok great! The changes in here are not new, I refactored a couple to stop passing data keys to components but that is is, it was previously using that functionality. I did this with hopes of having a better way of telling a component that it needs to watch on data changes of another field. 

The transformers that are using data to set a default value (like transformIdentify) seems to be standard practice, it is just pulling the value from the cookie (if it exists) and setting the default identifier. The issue was we were setting `defaultOption` but that doesn't get read or used in components, so I updated here to set it directly on the data bag for it to work correctly.